dmsrc/sanitize.dm

```diff
@@ -0,0 +1,36 @@
+/*
+ * Takes in a string and json_encode()"d lists to produce a sanitized string.
+ * This function operates on whitelists, there is currently no way to blacklist.
+ * Args:
+ * * text: the string to sanitize.
+ * * attribute_whitelist_json: a json_encode()'d list of HTML attributes to allow in the final string.
+ * * tag_whitelist_json: a json_encode()'d list of HTML tags to allow in the final string.
```

Comment:
The interface should take a list and json_encode in itself.

Reply:
I didn't do this so that you can store pre-encoded global lists to save on perf.

Reply:
Wouldn't that mean it's encoding on every call? The thing is that this will likely be called many times with only one or a few lists, so this introduces extra overhead.